packages/perseus-editor/src/widgets/interaction-editor/interaction-editor.tsx

```diff
@@ -222,11 +222,16 @@ class InteractionEditor extends React.Component<Props, State> {
                 <ElementContainer title="Grid settings">
                     <GraphSettings
                         editableSettings={["canvas", "graph"]}
-                        box={this.props.graph.box}
+                        box={this.props.graph.box as [number, number]}
                         labels={this.props.graph.labels}
-                        range={this.props.graph.range}
-                        step={this.props.graph.tickStep}
-                        gridStep={this.props.graph.gridStep}
+                        range={
+                            this.props.graph.range as [
+                                [number, number],
+                                [number, number],
+                            ]
+                        }
+                        step={this.props.graph.tickStep as [number, number]}
+                        gridStep={this.props.graph.gridStep as [number, number]}
```

Comment:
**suggestion**: In the next (or near future) PR you could change this components props so that they use tuples instead of `ReadonlyArray<>`. That'd eliminate the need to do the casting you're doing here... and I think it's more accurate.

Reply:
I think the props are already tuples. The problem is that we're doing a `slice()` on them [just above this](https://github.com/Khan/perseus/blob/3110c79bb524c78dd7ac0c737bb749f79b64f109/packages/perseus-editor/src/components/graph-settings.tsx#L303-L307), which is causing it to become `number[]` types. And then I get a type error because that means it can be shorter than the tuple expects.

Reply:
I can move the `as [number, number]` to those lines instead?

Reply:
Actually, all those `.slice()` calls look like they're just used to copy the array.  If you switch to spreading the array, Typescript seems to retain that their a tuple and you can eliminate the `as` casting entirely. 

https://www.typescriptlang.org/play?#code/C4TwDgpgBAggRgYygXigbwFBW1ATgQwDsBzCALigG1CBXAWzglwBopaGmBdAbgwF9eGBAHtCAZ2BQAHhXhJUmHHiKkKlAIwAGVls78MQ0RKghZiFOiw4CJclQB0jqfZuk9fIA

Note: you are welcome to land this PR as is and do these changes in a new PR.  Whatever works for you. :)